src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -218,14 +220,24 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, https_pr
     config_dp_endpoint = get_config_dp_endpoint(cmd, location)
 
     # Retrieving Helm chart OCI Artifact location
-    registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
+    if container_registry_repository:
+        if disable_auto_upgrade is False:
+            auto_upgrade_error = 'Disable auto upgrade when using custom repository'
+            telemetry.set_exception(exception=auto_upgrade_error, fault_type=consts.Custom_Repository_Disable_Auto_Upgrade_Fault_Type, summary=auto_upgrade_error)
+            raise ValidationError(auto_upgrade_error, recommendation='use --disable_auto_upgrade')
+
+        registry_path = "{}/azurearck8s/azure-arc-k8sagents:1.3.8".format(container_registry_repository)
+        # Send private registry information to telemetry
+        telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.PrivateContainerRegistry': container_registry_repository})
```

Comment:
Please also follow the step 5 in this [TSG](https://microsoft-my.sharepoint.com/personal/raunakp_microsoft_com/_layouts/15/Doc.aspx?sourcedoc={a6bd2448-96a4-40d7-bb82-2a99e7afba06}&action=edit&wd=target%28Details.one%7Ca9ccbe07-39fb-4714-8933-2c824af8e5e5%2FAdding%20custom%20telemetry%20in%20connectedk8s%20CLI%7C4be001fb-5d1b-44cb-ad95-7068617dca16%2F%29&wdorigin=703) to do the GDPR classification of this property once the changes are merged, otherwise we would not be able to collect this telemetry

Reply:
nit: doesn't impact flow, but terminology wise let's use CustomContainerRegistry instead of PrivateContainerRegistry - the registry itself could be public with anonymous access. so code readability wise it'll be clearer for new folks to the CLI codebase

Reply:
changed PrivateContainerRegistry to CustomContainerRegistry. will complete the TSG step before release